spirv_msl.cpp

```diff
@@ -2958,6 +3102,22 @@ string CompilerMSL::to_function_args(uint32_t img, const SPIRType &imgtype, bool
 		farg_str += to_expression(sample);
 	}
 
+	if (msl_options.swizzle_texture_samples && imgtype.image.sampled == 1 && imgtype.image.dim != DimBuffer &&
+	    (!is_gather || !imgtype.image.depth))
+	{
+		// Add the swizzle constant from the swizzle buffer.
+		if (!is_gather)
+			farg_str += ")";
+		// Get the original input variable for this image.
+		uint32_t img_var = img;
+		if (meta[img].image)
+			img_var = meta[img].image;
+		if (auto *var = maybe_get_backing_variable(img_var))
+			img_var = var->self;
+		farg_str += ", spvSwizzleConst[" +
+		            convert_to_string(get_metal_resource_index(get<SPIRVariable>(img_var), SPIRType::Image)) + "]";
```

Comment:
For sampling inside functions, I think this won't work. We won't know this information, and we might have to pass down swizzles to leaf functions, just like we sometimes have to pass down combined samplers to leaf functions in MSL.